src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1108,6 +1108,322 @@ void MFEMSidreDataCollection::RegisterField(const std::string& field_name,
   DataCollection::RegisterField(field_name, gf);
 }
 
+// private method
+void MFEMSidreDataCollection::
+addQuadratureFunction(const std::string &field_name, mfem::QuadratureFunction *qf,
+                     const std::string &buffer_name,
+                     axom::sidre::IndexType offset)
+{
+   sidre::Group* grp = nullptr;
+   if( m_bp_grp->hasGroup("fields/" + field_name))
+   {
+      grp = m_bp_grp->getGroup("fields/" + field_name);
+   }
+   else
+   {
+      SLIC_WARNING("Qfield " << field_name << " does not exist");
+   }
+
+   const int numDofs = qf->Size();
+
+   if(qf->GetData() == nullptr) {
+      AllocNamedBuffer(buffer_name, offset + numDofs);
+   }
+
+   // Make sure we have the View "values".
+   sidre::View *vv = alloc_view(grp, "values");
+
+   // Describe and apply the "values" View.
+   // If the data store has buffer for field_name (e.g. AllocNamedBuffer was
+   // called, or it was loaded from file), use that buffer.
+   sidre::View *bv = nullptr;
+   if (named_buffers_grp()->hasView(buffer_name))
+   {
+     bv = named_buffers_grp()->getView(buffer_name);
+   }
+
+   if (bv)
+   {
+      SLIC_ASSERT_MSG(bv->hasBuffer() && bv->isDescribed(), "");
+
+      // named buffers always have offset 0
+      SLIC_ASSERT_MSG(bv->getSchema().dtype().offset() == 0, "");
+      SLIC_ASSERT_MSG(bv->getNumElements() >= offset + numDofs, "");
+
+      if (vv->isEmpty())
+      {
+         vv->attachBuffer(bv->getBuffer())
+         ->apply(sidre::DOUBLE_ID, numDofs, offset);
+      }
+
+      qf->NewDataAndSize(vv->getData(), numDofs);
+   }
+   else
+   {
+      // If we are not managing the grid function's data,
+      // create a view with the external data
+      vv->setExternalDataPtr(sidre::DOUBLE_ID, numDofs, qf->GetData());
+   }
+   SLIC_ASSERT_MSG((numDofs > 0 && vv->isApplied()) ||
+                   (numDofs == 0 && vv->isEmpty() && vv->isDescribed()),
+                   "invalid View state");
+   SLIC_ASSERT_MSG(numDofs == 0 || vv->getData() == qf->GetData(),
+                   "View data is different from QuadratureFunction data");
+   SLIC_ASSERT_MSG(vv->getNumElements() == numDofs,
+                   "View size is different from QuadratureFunction size");
+}
+
+// private method
+// Should only be called on mpi rank 0 ( or if serial problem ).
+void MFEMSidreDataCollection::
+RegisterQFieldInBPIndex(const std::string& field_name, mfem::QuadratureFunction *qf)
```

Comment:
This looks pretty similar to `RegisterFieldInBPIndex` - it looks like the only thing needed here from the MFEM object (gridfunc/qfunc) is the vector dimension, so maybe that could replace the second parameter:

```cpp
RegisterFieldInBPIndex(const std::string& field_name, const int num_components)
```